XmpCore/Impl/XmpMetaParser.cs

```diff
@@ -106,13 +106,13 @@ public static XDocument Extract(byte[] bytes, ParseOptions options = null)
 
         private static IXmpMeta ParseXmlDoc(XDocument document, ParseOptions options)
         {
-            // sort node Attributes to match Java
-            // (keep namespace declarations in the order they are, and then prefixed names after in prefix:localname order)
+            // sort node Attributes to match Java DocumentBuilder (attribute order isn't supposed to matter but DocumentBuilder does some sorting)
+            // namespace declarations come first, and then all are sorted in prefix:localname order
             foreach(var node in document.Descendants().Where(d => d.Attributes().Count() > 1))
             {
-                var declarations = node.Attributes().Where(n => n.IsNamespaceDeclaration);
-                var orderedattribs = declarations.Concat(node.Attributes().Where(n => !n.IsNamespaceDeclaration).OrderBy(s => (node.GetPrefixOfNamespace(s.Name.Namespace) ?? "") + ":" + s.Name.LocalName));
-
+                var orderedattribs = node.Attributes()
+                                        .OrderBy(n => !n.IsNamespaceDeclaration)
+                                        .ThenBy(s => (node.GetPrefixOfNamespace(s.Name.Namespace) ?? "") + ":" + s.Name.LocalName);
```

Comment:
Could this be?

    .ThenBy(s => node.GetPrefixOfNamespace(s.Name.Namespace))
    .ThenBy(s => s.Name.LocalName);

It reads a little nicer and avoids creating _O(N)_ temporary strings (though it will allocate _O(1)_ object(s) for the additional `ThenBy` call).

If sorting of `null` and empty string isn't the same then you'd need the `?? ""` put back against the result of `GetPrefixOfNamespace`.